### PR TITLE
Add batch of unary functions

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64323 Add embedding shape analysis
* #64121 Max Pool with indices
* #63530 Add Maxpool to shape analysis / Opinfo
* #64300 Add symbolic shape comparison optimization
* #64299 Refactor to use shape arguments
* #63516 Add view with negative dim
* #63615 Generalize expand logic
* #63407 Add permute, arange
* #63365 Add support for slice, selec twith int, index_select
* #63099 Add squeeze, unsqueeze, transpose shape functins
* **#63050 Add batch of unary functions**

Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

### CODE REVIEW

File: torch/csrc/jit/runtime/symbolic_shape_registry.cpp [hunk 1]
```diff
@@ -308,11 +309,21 @@ static const OperatorMap<std::string>& get_schema_to_function_graph() {
...
       {"aten::add.Tensor(Tensor self, Tensor other, *, Scalar alpha=1) -> Tensor", "broadcast_one_unused_input"},
       {"aten::add.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> Tensor", "unary_two_unused_inputs"},
       {"aten::hardtanh(Tensor self, Scalar min_val=-1, Scalar max_val=1) -> Tensor", "unary_two_unused_inputs"},
       {"aten::adaptive_avg_pool2d(Tensor self, int[2] output_size) -> Tensor", "adaptive_avg_pool2d"},
+      {"aten::gelu(Tensor self) -> Tensor", "unary"},
+      {"aten::tanh(Tensor self) -> Tensor", "unary"},
+      {"aten::erf(Tensor self) -> (Tensor)", "unary"},
+      {"aten::zeros(int[] size, *, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None) -> (Tensor)", "unary_four_unused_inputs"},
+      {"aten::to.dtype(Tensor(a) self, int dtype, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor(a))", "unary_four_unused_inputs"},
+      {"aten::layer_norm(Tensor input, int[] normalized_shape, Tensor? weight=None, Tensor? bias=None, "
```
Comment (eellison) [inline torch/csrc/jit/runtime/symbolic_shape_registry.cpp:324]: image

image


layer norm and softmax dont have opinfos yet but they are extremely straightforward for shape analysis

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -6620,6 +6618,7 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCPU=floating_types_and(torch.bfloat16),
            dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
            sample_inputs_func=sample_inputs_softmax_variant,
+           assert_jit_shape_analysis=True,
```
Comment (ZolotukhinM) [inline torch/testing/_internal/common_methods_invocations.py:6621]: It seems that you've added more shape functions than tests. Is it intentional?

**Review by ZolotukhinM — APPROVED** [review]
Looks good!

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (eellison) [reply torch/testing/_internal/common_methods_invocations.py:6621]: yea, see my other comment about softmax & layernorm being straightforward unary ops. happy to expand opinfo coverage for more complicate operators but would prefer not to do it for unary ops.

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30347513).

**Review by priyaramani — APPROVED** [review]

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30347513).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30347513).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30347513).

**Pull request closed without merging**